DQMOffline/L1Trigger/src/L1TStage2CaloLayer2Offline.cc

```diff
@@ -386,17 +380,24 @@ void L1TStage2CaloLayer2Offline::fillJets(edm::Event const& e, const unsigned in
   double recoEta = leadingRecoJet.eta();
   double recoPhi = leadingRecoJet.phi();
 
-  double l1Et = closestL1Jet.et();
-  double l1Eta = closestL1Jet.eta();
-  double l1Phi = closestL1Jet.phi();
+  double l1Et = foundMatch ? closestL1Jet.et() : 0;
+  double l1Eta = foundMatch ? closestL1Jet.eta() : 9999;
+  double l1Phi = foundMatch ? closestL1Jet.phi() : 9999;
 
   // if no reco value, relative resolution does not make sense -> sort to overflow
   double outOfBounds = 9999;
   double resolutionEt = recoEt > 0 ? (l1Et - recoEt) / recoEt : outOfBounds;
-  double resolutionEta = std::abs(recoEta) > 0 ? (l1Eta - recoEta) / recoEta : outOfBounds;
-  double resolutionPhi = std::abs(recoPhi) > 0 ? (l1Phi - recoPhi) / recoPhi : outOfBounds;
+  double resolutionEta = l1Eta - recoEta;
+  double resolutionPhi = l1Phi - recoPhi;
 
   using namespace dqmoffline::l1t;
+  // fill efficiencies regardless of matched jet found
+  fillJetEfficiencies(recoEt, l1Et, recoEta);
+  // don't fill anything else if no matched L1 jet is found
+  if (!foundMatch){
+    return;
+  }
+
   // eta
   fill2DWithinLimits(h_L1JetEtavsCaloJetEta_, recoEta, l1Eta);
   fillWithinLimits(h_resolutionJetEta_, resolutionEta);
```

Comment:
The offline control plots could be filled even if there is no matching offline object found.

Reply:
Control plots as in `h_controlPlots_` or also `h_L1JetEtavsCaloJetEta_`, `h_resolutionJetEta_`, etc?